Source/Core/DiscIO/VolumeDirectory.cpp

```diff
@@ -25,7 +25,7 @@
 namespace DiscIO
 {
 static u32 ComputeNameSize(const File::FSTEntry& parent_entry);
-static std::string ASCIIToLowercase(std::string str);
+static std::string ASCIIToUppercase(std::string str);
```

Comment:
> Are there any hardware tests with filenames containing numbers?

What would such a hardware test be supposed to do? There's no FST parsing hardware, and I can't think of any part of the emulated software that relies on a particular ordering.

> The answer could probably be as simple as `strcasecmp` or `strverscmp` with the appropriate locale?

From the documentation of `strcasecmp` (which apparently quotes the POSIX standard):

              When the LC_CTYPE category of the locale being used is from
              the POSIX locale, these functions shall behave as if the
              strings had been converted to lowercase and then a byte
              comparison performed.  Otherwise, the results are unspecified.

So `strcasecmp` would behave like master, not like what ds84182 was looking for. Besides, is there a way to set the locale in a way that doesn't affect the rest of Dolphin, and do these functions exist on Windows at all?